Firestore/core/src/firebase/firestore/model/transform_operations.h

```diff
@@ -51,36 +53,31 @@ class TransformOperation {
   /** Returns the actual type. */
   virtual Type type() const = 0;
 
-// TODO(mikelehen): apply methods require FSTFieldValue which is Obj-C only.
-#if defined(__OBJC__)
   /**
    * Computes the local transform result against the provided `previousValue`,
    * optionally using the provided localWriteTime.
    */
-  virtual FSTFieldValue *applyToLocalView(
-      FSTFieldValue *previousValue, FIRTimestamp *localWriteTime) const = 0;
+  virtual FSTFieldValue* ApplyToLocalView(
+      FSTFieldValue* previousValue, FIRTimestamp* localWriteTime) const = 0;
 
   /**
    * Computes a final transform result after the transform has been acknowledged
    * by the server, potentially using the server-provided transformResult.
    */
-  virtual FSTFieldValue *applyToRemoteDocument(
-      FSTFieldValue *previousValue, FSTFieldValue *transformResult) const = 0;
-#endif
+  virtual FSTFieldValue* ApplyToRemoteDocument(
+      FSTFieldValue* previousValue, FSTFieldValue* transformResult) const = 0;
 
   /** Returns whether the two are equal. */
-  virtual bool operator==(const TransformOperation &other) const = 0;
+  virtual bool operator==(const TransformOperation& other) const = 0;
```

Comment:
What version of clang-format do you have?  Mine liked the spaces the other way...

clang-format version 7.0.0 (tags/google/stable/2018-01-11)

Reply:
I have:
```
clang-format version 7.0.0 (tags/google/stable/2018-01-11)
```
This is one of the few things for which clang-format is non-prescriptive. You can either bind left or bind right but not both within a file. Google's convention is right for Objective-C and left for C++. If you start writing a file the wrong way clang-format seems like it won't necessarily fix it :-(.

Reply:
Oh, wow. I didn't know it ever used conventions from the file to decide how to format the rest of the file. :-/